packages/beagle-react/cli/init/index.js

```diff
@@ -19,15 +22,23 @@ if (!fs.existsSync(dir)) {
 
   await fsPromises.writeFile('src/beagle/beagle-service.ts', beagleContent, fileOptions)
 
-  rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', (answer) => {
-    if (`${answer}` === 'y') {
-      fs.writeFile('src/App.tsx', appContent, function (err) {
-        if (err) throw err
-        process.exit()
-      })
-    }
-    else {
+  if (!fs.existsSync(dir + '/app.tsx')) {
+    fs.writeFile(dir + '/app.tsx', appContent, function (err) {
+      if (err) throw err
       process.exit()
-    }
-  })
+    })
+  }
+  else {
+    rl.question('Do you want to replace "app.tsx" content with the Beagle configuration (y or n)?', (answer) => {
+      if (`${answer}` === 'y') {
+        fs.writeFile('src/app.tsx', appContent, function (err) {
+          if (err) throw err
+          process.exit()
+        })
+      }
```

Comment:
Using the tips that @Tiagoperes gave to you, about `fs/promises`, you can also abstract those `fs.writeFile` into a method that returns a `Promise` and handle those `throw`, because they are currently uncaught.

```suggestion
async function createAppTsx(path: string, content: string): Promise<void> {
  try {
    await writeFile(`${path}/app.tsx`, content)
  } catch (error) {
    console.error(error)
    process.exit()
  }
}

// using it
// { ... }
if (`${answer}` === 'y') {
    await createAppTsx('src', appContent)        
}
process.exit()
```

Reply:
PS: do not commit this suggestion... made it a suggestion to be more easy to visualize